src/Api.js

```diff
@@ -86,37 +107,70 @@ class Api {
 		// This is better than using the 'oldid' URL parameter directly, because it takes into
 		// account the 'direction' URL parameter.
 		revId = this.mwConfig.get( 'wgRevisionId' );
-		curRevId = this.mwConfig.get( 'wgCurRevisionId' );
-		if ( revId !== curRevId ) {
-			parts.push( revId );
-		}
+		// Always include the revision ID, to make sure we are always asking for
+		// the correct revision, whether the page was just edited, or, whether the
+		// page was edited by someone else while we were looking at the current page
+		parts.push( revId );
 
 		// Compile the full URL.
 		return parts.join( '/' ) + '/';
 	}
 
 	/**
-	 * Get the WikiWho replacement for `.mw-parser-output` HTML.
-	 * @return {string}
+	 * Get a unique identifier for a specific promise.
+	 * The promise is unique if the revision ID is the same, and should
+	 * change (and hence force a refetch) if the revision ID changed, like
+	 * in the case of looking at the page after a VE save (without refresh)
+	 *
+	 * @return {string} Identifier for this promise
 	 */
-	getReplacementHtml() {
-		return this.results.extended_html;
+	getContentRevisionIdentifier() {
```

Comment:
I was going to suggest that this method be used above in `getAjaxURL()`, but maybe it's meant to be a *cache* identifier, not a revision identifier (I mean, they might be the same thing of course). If it is then maybe rename this method, and if it's not then it can be used in both places and the `mwConfig.get( 'wgRevisionId' )` only occur here.

Reply:
Done.